driver.go

```diff
@@ -159,8 +165,8 @@ func connect(opts *Options) (*Conn, error) {
 
 	httpTransport, ok := transport.(*thrift.THttpClient)
 	if ok {
-		// TODO: currently masking as a python connector until additional user agents are white listed.
-		httpTransport.SetHeader("User-Agent", "godatabrickssqlconnector/0.9.0 (go-dbsql)")
+		userAgent := fmt.Sprintf("godatabrickssqlconnector/0.9.0 (%s)", opts.ClientId)
```

Comment:
@mattdeekay I wonder if you are planning to also add support for adding a user-agent suffix (passed by the end user) or is that outside of the scope of this PR?

Reply:
I was thinking that the `clientId` would be the user-agent suffix within parenthesis that the end user could override or else would default to "go-dbsql". Is this an anti-pattern - should I try something else?

Reply:
Does it make sense to consider the suffix separately? Moe it'd also help with linking to a JIRA ticket that provides context.

I assume this code change is to make sure we will support those existing patterns.

https://src.dev.databricks.com/databricks/universe@master/-/blob/common/logging/redactor/test/DatabricksLogRedactorSuite.scala?L1019-L1039

Reply:
@yunbodeng-db of course. It totally makes sense and I was not saying we have to do this now. I was just asking about the scope of the PR. :-)

Reply:
I'm not able to see the above link but for the scope of this PR, would Go connector version + clientId suffice? I can create a separate work item for any other changes (suffix, os version, golang version) we'd want but what's in this PR seems to be at feature parity with Python SDK

Reply:
agreed @mattdeekay let's do other things outside of the scope of this PR. Thanks